Use htmx for loading search results

Simrayz opened the pull request:
## Scope and purpose

This PR changes info search forms to use HTMX to fetch and display search results. This avoids refreshing the entire page on each search, and gives a smoother user experience.

This PR is also a follow-up to https://github.com/Uninett/nav/pull/3563, where the CSRF tokens in these search forms were overlooked. The CSRF tokens are removed by using `disable_csrf=True` in the `info/_search_form.html` partial.

## How to test

1. Open the "Search" tool, `/search`
2. Verify that a search returns results without refreshing the page. E.g. "100" should return Rooms, Interfaces, Vlans and Unrecognized neighbors.
3. Click "IP Device Search", and verify that the results are loaded without refreshing.
4. Repeat the same process for Room, Location, VLAN, Prefix and Device group search.

## Contributor Checklist

Every pull request should have this checklist filled out, no matter how small it is.
More information about contributing to NAV can be found in the
[Hacker's guide to NAV](https://nav.readthedocs.io/en/latest/hacking/hacking.html#hacker-s-guide-to-nav).

<!-- Add an "X" inside the brackets to confirm -->
<!-- If not checking one or more of the boxes, please explain why below each or remove the line if not applicable. -->

* [x] Added a changelog fragment for [towncrier](https://nav.readthedocs.io/en/latest/hacking/hacking.html#adding-a-changelog-entry)
* [x] Added/amended tests for new/changed code
* [ ] Added/changed documentation
* [x] Linted/formatted the code with ruff, easiest by using [pre-commit](https://nav.readthedocs.io/en/latest/hacking/hacking.html#pre-commit-hooks-and-ruff)
* [x] Wrote the commit message so that the first line continues the sentence "If applied, this commit will ...", starts with a capital letter, does not end with punctuation and is 50 characters or less long. See https://cbea.ms/git-commit/
* [x] Based this pull request on the correct upstream branch: For a patch/bugfix affecting the latest stable version, it should be based on that version's branch (`<major>.<minor>.x`). For a new feature or other additions, it should be based on `master`.
* [ ] If applicable: Created new issues if this PR does not fix the issue completely/there is further work to be done
* [ ] If it's not obvious from a linked issue, described how to interact with NAV in order for a reviewer to observe the effects of this change first-hand (commands, URLs, UI interactions)
* [ ] If this results in changes in the UI: Added screenshots of the before and after
* [ ] If this adds a new Python source code file: Added the [boilerplate header](https://nav.readthedocs.io/en/latest/hacking/hacking.html#python-boilerplate-headers) to that file

<!-- Make this a draft PR if the content is subject to change, cannot be merged or if it is for initial feedback -->

johannaengland reviewed (changes requested):
When I do a location search it still adds the CSRF token to the url

And three things I noticed during my review, but that were not introduced by this PR:

When I do a device group search I get the following error in the console:

```
Could not find needed elements multiple_select.js:35:21
    MultipleSelect http://localhost:8000/static/js/src/plugins/multiple_select.js?bust=5.14.2.dev160+g437d239fe:35
    <anonymous> http://localhost:8000/static/js/src/info_netboxgroup.js?bust=5.14.2.dev160+g437d239fe:11
    jQuery 5
    <anonymous> http://localhost:8000/static/js/src/info_netboxgroup.js?bust=5.14.2.dev160+g437d239fe:10
    execCb http://localhost:8000/static/js/libs/require.js:31
    check http://localhost:8000/static/js/libs/require.js:21
    enable http://localhost:8000/static/js/libs/require.js:25
    s http://localhost:8000/static/js/libs/require.js:8
    emit http://localhost:8000/static/js/libs/require.js:26
    q http://localhost:8000/static/js/libs/require.js:7
    emit http://localhost:8000/static/js/libs/require.js:26
    check http://localhost:8000/static/js/libs/require.js:22
    enable http://localhost:8000/static/js/libs/require.js:25
    s http://localhost:8000/static/js/libs/require.js:8
    emit http://localhost:8000/static/js/libs/require.js:26
    q http://localhost:8000/static/js/libs/require.js:7
    emit http://localhost:8000/static/js/libs/require.js:26
    check http://localhost:8000/static/js/libs/require.js:22
    enable http://localhost:8000/static/js/libs/require.js:26
    init http://localhost:8000/static/js/libs/require.js:19
    V http://localhost:8000/static/js/libs/require.js:17
    completeLoad http://localhost:8000/static/js/libs/require.js:30
    onScriptLoad http://localhost:8000/static/js/libs/require.js:32
    (Async: EventListener.handleEvent)
    load http://localhost:8000/static/js/libs/require.js:33
    load http://localhost:8000/static/js/libs/require.js:31
    load http://localhost:8000/static/js/libs/require.js:21
    fetch http://localhost:8000/static/js/libs/require.js:20
    check http://localhost:8000/static/js/libs/require.js:22
    enable http://localhost:8000/static/js/libs/require.js:26
    enable http://localhost:8000/static/js/libs/require.js:29
    enable http://localhost:8000/static/js/libs/require.js:25
    s http://localhost:8000/static/js/libs/require.js:8
    q http://localhost:8000/static/js/libs/require.js:7
    enable http://localhost:8000/static/js/libs/require.js:25
    init http://localhost:8000/static/js/libs/require.js:19
    V http://localhost:8000/static/js/libs/require.js:17
    completeLoad http://localhost:8000/static/js/libs/require.js:30
    onScriptLoad http://localhost:8000/static/js/libs/require.js:32
    (Async: EventListener.handleEvent)
    load http://localhost:8000/static/js/libs/require.js:33
    load http://localhost:8000/static/js/libs/require.js:31
    load http://localhost:8000/static/js/libs/require.js:21
    fetch http://localhost:8000/static/js/libs/require.js:20
    check http://localhost:8000/static/js/libs/require.js:22
    enable http://localhost:8000/static/js/libs/require.js:26
    enable http://localhost:8000/static/js/libs/require.js:29
    enable http://localhost:8000/static/js/libs/require.js:25
    s http://localhost:8000/static/js/libs/require.js:8
    q http://localhost:8000/static/js/libs/require.js:7
    enable http://localhost:8000/static/js/libs/require.js:25
    init http://localhost:8000/static/js/libs/require.js:19
    require http://localhost:8000/static/js/libs/require.js:29
    requirejs http://localhost:8000/static/js/libs/require.js:11
    <anonymous> http://localhost:8000/static/js/src/info_netboxgroup.js?bust=5.14.2.dev160+g437d239fe:1
```

and in the resulting table the table header and table contents are not aligned (see "Devices" column):

<img width="1047" height="162" alt="image" src="https://github.com/user-attachments/assets/adde46b4-0c13-4c4d-876f-0601bedb78ea" />

and the width of the table changes between initial load and then the search - before:

<img width="1678" height="162" alt="image" src="https://github.com/user-attachments/assets/3f39216f-0c77-4e8c-99b3-f569d09bb9f2" />

Simrayz commented:
> When I do a location search it still adds the CSRF token to the url

Thanks! I was certain I updated all the search forms. Fixed now.

> and in the resulting table the table header and table contents are not aligned (see "Devices" column):
and the width of the table changes between initial load and then the search - before:

I adjusted the device groups table slightly to make the layout more consistent, by making the description column take up all available space.

Simrayz commented:
I can take a look at `multiple_select`. I'm not sure what the plugin is supposed to do, the bug is present in 5.13.x as well. I think we can just remove the MultipleSelect plugin completely, as this is the only place it is referenced, but not used.

lunkwill42 reviewed (changes requested):
Looks nice, but there is now an apparent lack of feedback when the search is running slowly. It seems like the page just locks up, when probably a spinner should appear...

Simrayz commented:
> Very good idea with the loading indicator!
> 
> When I search for IP devices and am entering a search term that does not produce results, then I am forwarded to `http://localhost:8000/ipdevinfo/abc/`:
> 
> <img alt="image" width="433" height="843" src="https://private-user-images.githubusercontent.com/39082189/508942136-d8129c25-73f2-4a74-ade6-964352639c5e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjIyNTUyNzIsIm5iZiI6MTc2MjI1NDk3MiwicGF0aCI6Ii8zOTA4MjE4OS81MDg5NDIxMzYtZDgxMjljMjUtNzNmMi00YTc0LWFkZTYtOTY0MzUyNjM5YzVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTA0VDExMTYxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUwMTg3YWRkNTAxM2VhMmEwYzIyZjIyYWZmMmRkODI2ZGQyODEzN2YzM2U2OTIxMDZkZDEyZDkwYjI5ZDRkMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tNYPDVdYLtbB14hOsyUOjjBoYzqODCr0EiHuC05ZMng">

This is how it was from before, it uses some "FallbackProvider" that shows this page. Which is not very helpful, but that's how it is 🤷‍♂️

lunkwill42 reviewed (approved):
Loading indicator is there now, so I guess I'm happy 😄 🎄
